contracts/pool/pools/liquid/LiquidPool.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "../TokenPoolBase.sol";
+import "./WithdrawalEligibility.sol";
+
+error ExistentTokensToClaim();
+
+/**
+ * @title LiquidPool
+ * @dev Provides functionality for staking, locking, and withdrawing tokens
+ * Tokens can be withdrawn with respecting grace period or immediately by paying ignore fee
+ */
+contract LiquidPool is TokenPoolBase, WithdrawalEligibility {
+    /**
+     * @notice Stakes tokens in the liquid pool
+     * @param amount The amount of tokens to stake
+     */
+    function liquidStake(
+        uint256 amount
+    ) external onlySufficientAmount(amount) nonReentrant {
+        Stake memory stakeInfo = stakes[msg.sender];
+
+        stakes[msg.sender] = Stake({
+            amount: stakeInfo.amount + amount,
+            eligibleWithdraw: stakeInfo.eligibleWithdraw
+        });
+
+        _claimTokens(amount);
+
+        emit TokenActivity(ActivityType.Stake, msg.sender, amount);
+    }
+
+    /**
+     * @notice Immediately withdraws staked tokens and pays grace period removal fee
+     * @param amount The amount of tokens to withdraw
+     */
+    function withdraw(
+        uint256 amount
+    ) external onlySufficientAmount(amount) nonReentrant {
+        Stake memory stakeInfo = stakes[msg.sender];
+
+        uint256 availableToWithdraw = stakeInfo.amount;
+
+        _checkAvailableWithdraw(availableToWithdraw, amount);
+
+        uint256 amountToWithdraw = _substractRemovalFee(amount);
+
+        stakeInfo.eligibleWithdraw = _initEligibleWithdraw();
+        stakeInfo.amount = availableToWithdraw - amount;
+
+        stakes[msg.sender] = stakeInfo;
+
+        _releaseTokens(amountToWithdraw);
+
+        emit TokenActivity(ActivityType.Withdraw, msg.sender, amountToWithdraw);
+    }
+
+    /**
+     * @notice Withdraws staked tokens after starting grace period
+     * @param ignoreGracePeriod Whether to ignore the grace period after starting it
+     */
+    function withdraw(bool ignoreGracePeriod) external nonReentrant {
```

Comment:
Kad user ne zeli cekat grace period end time, onda moze i zvat `withdraw(amount)` funkciju tj. ne triba zvat ovu `withdraw(true)`. Mozda ovde pretpostavit da se ne ignorira grace period, ili eliminirat gornju `withdraw(amount)` funkciju tako da user mora inicirat withdraw ali moze withdrawat i bez cekanja grace perioda (ako to stima prema specifikaciji). Isto je i za unlockanje. Ali i nije toliko ovo bitno.

Reply:
ovu sam stavia bas iz razloga da u slucaju ako user pokrene grace period da ga moze ignorirat ali da kad bolje razmislim sada ima smisla sta govoris moze pozvat gornju i skippat grace period. Dodat cu tako da u ovom slucaju se podrazumjeva da nezeli ignorirat grace period